src/Block/BreadcrumbBlockService.php

```diff
@@ -41,7 +41,7 @@ class BreadcrumbBlockService extends BaseBreadcrumbMenuBlockService
      * @param string $context
      * @param string $name
      */
-    public function __construct($context, $name, EngineInterface $templating, MenuProviderInterface $menuProvider, FactoryInterface $factory, CmsManagerSelectorInterface $cmsSelector)
+    public function __construct($context, $name, Environment $templating, MenuProviderInterface $menuProvider, FactoryInterface $factory, CmsManagerSelectorInterface $cmsSelector)
```

Comment:
@VincentLanglet, this class should be final?

other question, @Daric971 should provider some BC compatibility?

Reply:
`@final since sonata-project/page-bundle 3.26` but all the final annotation can be done in another PR.